src/history/cam_hist_file.F90

```diff
@@ -0,0 +1,2162 @@
+           file=__FILE__, line=__LINE__-1)
+
+      call this%field_list_hash_table%initialize(num_fields)
+      allocate(this%field_list(num_fields), stat=ierr)
+      call check_allocate(ierr, subname, 'this%field_list',             &
+           file=__FILE__, line=__LINE__-1)
+
+      field_index = 1
+      ! Add the field names and associated accumulate types to the object
+      do idx = 1, num_inst_fields
+         this%accumulate_types(field_index) = 'lst'
+         this%field_names(field_index) = inst_fields(idx)
+         field_index = field_index + 1
+      end do
+      do idx = 1, num_avg_fields
+         this%accumulate_types(field_index) = 'avg'
+         this%field_names(field_index) = avg_fields(idx)
+         field_index = field_index + 1
+      end do
+      do idx = 1, num_min_fields
+         this%accumulate_types(field_index) = 'min'
+         this%field_names(field_index) = min_fields(idx)
+         field_index = field_index + 1
+      end do
+      do idx = 1, num_max_fields
+         this%accumulate_types(field_index) = 'max'
+         this%field_names(field_index) = max_fields(idx)
+         field_index = field_index + 1
+      end do
+      do idx = 1, num_var_fields
+         this%accumulate_types(field_index) = 'var'
+         this%field_names(field_index) = var_fields(idx)
+         field_index = field_index + 1
+      end do
+
+   end subroutine config_configure
+
+   ! ========================================================================
+
+   subroutine config_print_config(this)
+      use string_utils,   only: to_str
+      use spmd_utils,     only: masterproc
+      use cam_logfile,    only: iulog
+      use cam_abortutils, only: endrun
+      ! Dummy argument
+      class(hist_file_t), intent(in) :: this
+
+      if (masterproc) then
+         if (this%has_instantaneous) then
+            write(iulog, '(3a)') "Instanteous history configuration for volume = ",   &
+               trim(this%volume), 'i'
+         end if
+         if (this%has_accumulated) then
+            write(iulog, '(3a)') "Accumulated history configuration for volume = ",   &
+               trim(this%volume), 'a'
+         end if
+         select case(this%hfile_type)
+         case (hfile_type_history)
+            write(iulog, *) "File will contain model history (diagnostics) output"
+         case (hfile_type_init_value)
+            write(iulog, *) "File will contain values for model initialization"
+         case (hfile_type_sat_track)
+            write(iulog, *) "File will contain satellite track values"
+         case (hfile_type_restart)
+            write(iulog, *) "File contains history restart information"
+         case default
+            call endrun("ERROR: Unknown CAM history file type, "//            &
+                 to_str(this%hfile_type))
+         end select
+         if (this%rl_kind == REAL64) then
+            write(iulog, '(a)') "  Ouput precision, 64 bits"
```

Comment:
Just noting here and two lines below, a typo `Ouput` -> `Output`. Thanks!

Reply:
thanks @jimmielin ! typo fixed.